pkg/handler/collector/github/github.go

```diff
@@ -0,0 +1,140 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/guacsec/guac/pkg/handler/collector"
+	"github.com/guacsec/guac/pkg/handler/collector/file"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CollectorGitHubDocument = "GitHubCollector"
+)
+
+type githubDocumentCollector struct {
+	url           string
+	dir           string
+	lastChecked   time.Time
+	poll          bool
+	interval      time.Duration
+	fileCollector collector.Collector
+}
+
+func NewGitHubDocumentCollector(ctx context.Context, url string, dir string, poll bool, interval time.Duration) *githubDocumentCollector {
+	fileCollector := file.NewFileCollector(ctx, dir, false, time.Second)
+
+	return &githubDocumentCollector{
+		url:           url,
+		dir:           dir,
+		poll:          poll,
+		interval:      interval,
+		fileCollector: fileCollector,
+	}
+}
+
+func (g *githubDocumentCollector) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
+	// Replace with your own personal access token, export to use
+	token := os.Getenv("API_KEY")
+
+	// Replace with the owner and name of the repository
+	owner := "slsa-framework"
+	repo := "slsa-github-generator"
+
+	// Replace with the path of the directory where you want to download the assets
```

Comment:
Just a note it's often useful to add a comment like `TODO: ...` or `FIXME: ...` for these sorts of comments as you can easily search them while working so you know what you have to go back and fix, vs. what comments are just for readers to better understand what you're doing.